src/lib/legacy.ts

```diff
@@ -309,8 +309,18 @@ export async function fromV2TargetApps(
 						const appId = parseInt(id, 10);
 						const app = apps[appId];
 
-						// If local mode just use id as uuid
-						const uuid = local ? id : await getUUIDFromAPI(appId);
+						let uuid: string;
+						try {
+							// If local mode just use id as uuid
+							uuid = local ? id : await getUUIDFromAPI(appId);
+						} catch {
+							// If getUUIDFromAPI fails, set uuid to id.
+							// getUUIDFromAPI will fail with network connection errors,
+							// so when the device reconnects and sees a v3 target state,
+							// user containers will restart. Update locks should be leveraged
+							// during critical processes in container to avoid this.
+							uuid = id;
```

Comment:
Has unexpected behavior after the device connects to the internet post-preload. Will investigate

Reply:
Since we're resolving this by bumping balena-preload in balena CLI to preload v3 target state, this behavior isn't an issue anymore.